### ISSUE TITLE
E2E tests to cover multi-partition edge use cases

### ISSUE THREAD

**Issue by KeranYang** [numaproj/numaflow#763]
# Summary

Add E2E tests to cover multi-partition edge use cases, once we finish the feature implementation. The test could include:
* partition level watermark propagation verification
* partition level processing rate verification
* etc.

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by xdevxy** [comment]
Can you explain more one what kind of verification you are expecting for watermark propagation and processing rate? Thanks!

**Comment by KeranYang** [comment]
@xdevxy Yes, I added more details to the issue description on how I plan to implement the verification.
